core/cloudflow-streamlets/src/main/scala/cloudflow/streamlets/proto/javadsl/ProtoInlet.scala

```diff
@@ -39,10 +39,19 @@ final case class ProtoInlet[T <: GeneratedMessageV3](
 }
 
 object ProtoInlet {
-  // Java API
   def create[T <: GeneratedMessageV3](name: String, clazz: Class[T]): ProtoInlet[T] =
     ProtoInlet[T](name, clazz)
 
   def create[T <: GeneratedMessageV3](name: String, clazz: Class[T], hasUniqueGroupId: Boolean): ProtoInlet[T] =
     ProtoInlet[T](name, clazz, hasUniqueGroupId)
+
+  def create[T <: GeneratedMessageV3](
+      name: String,
+      clazz: Class[T],
+      hasUniqueGroupId: Boolean,
+      errorHandler: (Array[Byte], Throwable) => Optional[T]): ProtoInlet[T] =
+    ProtoInlet[T](name, clazz, hasUniqueGroupId, (a, t) => {
+      val opt = errorHandler(a, t)
+      if (opt.isPresent) Some(opt.get()) else None
```

Comment:
Isn't this same as returning the opt itself ?

Reply:
Oh no, wait, actually this is converted to the Scala API. The constructor takes Option, not Optional.